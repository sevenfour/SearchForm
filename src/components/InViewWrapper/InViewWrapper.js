import React from 'react';
import PropTypes from 'prop-types';

import clsx from 'clsx';

import { useInView } from 'react-intersection-observer';

import styles from './InViewWrapper.module.css';

const InViewWrapper = props => {

  const {
    className,
    content: Content,
    placeHolder: PlaceHolder,
    initialInView
  } = props;

  const [ref, inView] = useInView({initialInView});

  return (
    <div
      className={clsx(styles.container, className)}
      ref={ref}
    >
      {inView ? Content : PlaceHolder}
    </div>
  );
};

InViewWrapper.propTypes = {
  className: PropTypes.string,
  content: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.element,
  ]),
  initialInView: PropTypes.bool,
  placeHolder: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.element,
  ]),
  rootMargin: PropTypes.string
};

InViewWrapper.defaultProps = {
  initialInView: false
};

export default InViewWrapper;
