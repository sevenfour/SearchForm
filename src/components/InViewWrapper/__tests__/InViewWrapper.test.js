import React from 'react';
import { shallow } from 'enzyme';

import InViewWrapper from '..';

describe('<InViewWrapper /> component tests', () => {

  const content = (
    <div>
      content
    </div>
  );
  const placeHolder = (
    <div>
      placeHolder
    </div>
  );

  it('renders InViewWrapper', () => {
    const inView = shallow(
      <InViewWrapper
        content={content}
        placeHolder={placeHolder}
      />
    );

    expect(inView).toMatchSnapshot();
  });

});
