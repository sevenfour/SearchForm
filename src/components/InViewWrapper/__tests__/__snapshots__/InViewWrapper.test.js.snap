// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InViewWrapper /> component tests renders InViewWrapper 1`] = `
<div
  className="container"
>
  <div>
    placeHolder
  </div>
</div>
`;
